### PR TITLE
add command GEORADIUSBYMEMBER

### PR DESCRIPTION
**Pull request by kyeett**
Add the `GEORADIUSBYMEMBER` command as specified here:
https://redis.io/commands/georadiusbymember

Initial implementation to see if I am on the right track. 

I wasn't sure how to run the integration tests. I will have a look at it next.

### CODE REVIEW

File: cmd_geo.go [hunk 1]
```diff
@@ -370,6 +372,200 @@ func (m *Miniredis) cmdGeoradius(c *server.Peer, cmd string, args []string) {
...
+			return
+		}
+
+		// get position of member
+		if !db.ssetExists(key, member) {
+			c.WriteError("ERR could not decode requested zset member")
+			return
+		}
+		score := db.ssetScore(key, member)
+		longitude, latitude := fromGeohash(uint64(score))
```
Comment (kyeett) [inline cmd_geo.go]: A fair amount of code could be shared between GEORADIUS and GEORADIUSBYMEMBER, if needed.

The selected part is what is unique for GEORADIUSBYMEMBER.

File: cmd_geo_test.go [hunk 2]
```diff
@@ -289,3 +289,160 @@ func TestGeodist(t *testing.T) {
...
+		equals(t, "Catania", name2)
+		equals(t, 166.2742, dist2)
+
+		// in meter
+		res, err = redis.Values(c.Do("GEORADIUSBYMEMBER", "Sicily", "Catania", 200000, "m", "WITHDIST"))
+		ok(t, err)
+		equals(t, 2, len(res))
+		distance, err := redis.Float64(res[0].([]interface{})[1], nil)
+		ok(t, err)
+		equals(t, 166274.1514, distance) // in meter
```
Comment (kyeett) [inline cmd_geo_test.go]: When I try this in real redis, I get a slightly different value. For now I hardcoded the test to the result I get from `miniredis`

Real redis: 166274.1516
Miniredis : 166274.1514

Is this

File: cmd_geo.go [hunk 3]
```diff
@@ -370,6 +372,200 @@ func (m *Miniredis) cmdGeoradius(c *server.Peer, cmd string, args []string) {
...
+			}
+			withStoredist = true
+			storedistKey = args[0]
+			args = args[1:]
+		default:
+			setDirty(c)
+			c.WriteError("ERR syntax error")
+			return
+		}
+	}
```
Comment (kyeett) [inline cmd_geo.go]: Identical to second half of argument parsing for GEORADIUS. Should probably be shared.

File: integration/geo_test.go [hunk 4]
```diff
@@ -265,6 +265,156 @@ func TestGeoradius(t *testing.T) {
...
+			succRound3("GEORADIUSBYMEMBER", "stations", "Astor Pl", 400, "km", "ASC", "COUNT", 999),
+			fail("GEORADIUSBYMEMBER", "stations", "Astor Pl", 400, "km", "COUNT"),
+			fail("GEORADIUSBYMEMBER", "stations", "Astor Pl", 400, "km", "COUNT", 0),
+			fail("GEORADIUSBYMEMBER", "stations", "Astor Pl", 400, "km", "COUNT", -12),
+			fail("GEORADIUSBYMEMBER", "stations", "Astor Pl", 400, "km", "COUNT", "foobar"),
+
+			// non-existing key
+			// succ("GEORADIUSBYMEMBER", "foo", "Astor Pl", 4, "km"), // Failing
+			// geo_test.go:268: value error. expected: []interface {}{} got: <nil> case: main.command{cmd:"GEORADIUSBYMEMBER", args:[]interface {}{"foo", "Astor Pl", 4, "km"}, error:false, sort:false, loosely:false, errorSub:"", receiveOnly:false, roundFloats:0, closeChan:false}
+
```
Comment (kyeett) [inline integration/geo_test.go]: Failing test case, will have to fix.

File: integration/geo_test.go [hunk 5]
```diff
@@ -757,9 +907,23 @@ func TestGeo(t *testing.T) {
 		succRound3("GEORADIUS", "stations", -73.9718893, 40.7728773, 4, "km", "ASC", "COUNT", 3),
 		succRound3("GEORADIUS", "stations", -73.9718893, 40.7728773, 4, "km", "ASC", "COUNT", 99999),
 
+		succRound3("GEORADIUSBYMEMBER", "stations", "Astor Pl", 20, "km"),              // This testcase doesn't pass with 4 km. I had expected it to pass with 8km at least. 20km seems to do it. How can 4k below pass, when 4k here doesn't?
+		succRound3("GEORADIUSBYMEMBER", "stations", "Astor Pl", 20, "km", "WITHDIST"),  // This testcase doesn't pass with 4 km.
+		succRound3("GEORADIUSBYMEMBER", "stations", "Astor Pl", 20, "km", "WITHCOORD"), // This testcase doesn't pass with 4 km.
```
Comment (kyeett) [inline integration/geo_test.go]: These 3 testcases needed 20 km radius instead of 4 km. Not sure why.

Comment (alicebob) [reply integration/geo_test.go]: The order is undefined:
>The command default is to return unsorted items. Two different sorting methods can be invoked using the following two options:

I added some commands to sort results. Now they work fine with 4km. I guess that with 20Km the lookup order works out differently.

**Comment by alicebob** [comment]
It looks good to me. I'll have another look in a few days. Thanks!

**Comment by alicebob** [comment]
@kyeett, can you squash this in your branch so I can merge this? Thanks!

(If think if I quash you won't get credits for your contribution in github)

**Comment by kyeett** [comment]
@alicebob hopefully I did it the right way :-)

**Comment by alicebob** [comment]
Thanks! I'll make it a new release in a week or so.

**Pull request merged**